### ISSUE TITLE
Make checking for `application/x-zerosize` optional

### ISSUE THREAD

**Issue by Anomalocaridid** [ebassi/xdg-mime-rs#29]
I would like for checking for `application/x-zerosize` optional. This would be useful for when knowing the "intended" (for lack of better terms) type of the file is more important than its exact contents.

This could probably be accomplished by adding a boolean field to `GuessBuilder` and a method to set it that would be checked in `guess` when the length of the file is being checked.

I intend to do a PR for this myself, but since this affects the public API, I figured it would be a good idea to discuss what approach to take first.

**Comment by ebassi** [comment]
Something like:

```rust
pub fn zero_size(&mut self, allow: bool) -> &mut Self { ... }
```

and then using the buffer size only if `allow` is `true` (which would be the default, to preserve backward compatibility)?

**Comment by Anomalocaridid** [comment]
Yes, exactly like that.

**Comment by ebassi** [comment]
I think it's a perfectly fine addition; I just released 0.4.0 because of the nom 5 → 7 transition, but I'm happy to do a 0.5.0 release with new API.

**Comment by Anomalocaridid** [comment]
Solved by #30
